tests/Fixer/PhpUnit/PhpUnitNoExpectationAnnotationFixerTest.php

```diff
@@ -41,6 +41,32 @@ public function testFix($expected, $input = null, array $config = [])
     public function provideTestFixCases()
     {
         return [
+            'empty exception message' => [
+                '<?php
+    final class MyTest extends \PHPUnit_Framework_TestCase
+    {
+        /**
+         */
+        public function testFnc()
+        {
+            $this->setExpectedException(\FooException::class, \'\');
```

Comment:
Should be just `$this->setExpectedException(\FooException::class);`.

Reply:
@GrahamCampbell I don't know. What I can say is that is must not: it works this way for the code-base I had the error with.

Not dropping that annotation value also allowed me (as a user) to spot later on the problem that originally no message was given. The annotation that is getting transposed here is about looking for that string somewhere in the exception message - this makes hardly any sense with an empty string - so the user might want to be able to learn this its own w/o having the fixer hiding an issue.

So it already originally made no sense and in my case I didn't want to have the error, but it's fine if I can spot later on a useless exception expectation with a different inspection).

Just so it's better understandable why I kept the use-less looking behaviour in.

So please let me know if this lowers your change suggestion.

Reply:
The previous code with the empty annotation message was pointless, and should have been removed. It's obviously always going to be true that the empty string is a prefix of any string.

Therefore, we should remove the tautology check.

Reply:
Well then it is perhaps better to let the error in so that the user can actually notice the problem as CS-Fixer fails. This is the way it helped me actually. I don't like to remove a program flaw that hides away other issues. When it was the last line of defence, then it should be kept albeit the reporting is not very precise.

Reply:
I'm just saying that when this happens, the generated code should not include the quotes. Just setup the code with only one argument, not two.

Reply:
Like I said, I think the generated code should be:

```
$this->setExpectedException(\FooException::class);
```

Reply:
https://github.com/ktomk/PHP-CS-Fixer/pull/1

Reply:
Like I wrote, I see the benefit in keeping the error message then.I don't want to be the person who introduces the burden then for others to see their problem places getting hidden by a fixer that should just port.

But maybe there is a solution: What I asked myself is, is there any possibility that a fixer can signal problematic spots for the user? I also have other fixer-fails situations which were all good to know about, like with these not given (empty) exception messages which signal a problem (the message has been forgotten for example) a warning with a problem description (text), file and line, perhaps offset, could be given?

I would be very fine w/ not having the error in PHP-CS-Fixer and perhaps also trying best to fix the code-issue (like you want it) but to signal the user that a fishy spot has been detected.

This would also allow me to suggest more problem places I was running over, which are of similar matter. Having PHP-CS-Fixer failing with errors helped me to spot the places, but a better location which code triggered an issue would have helped me a lot.

But this might be bike-shedding and very much driven from my concrete needs due to the issues I was running over, so it's more the question if something like such a reporting facility already exists.